### DISCUSSION TITLE
Faster SDF calculation from mesh?

### DISCUSSION THREAD

**Discussion by FrankEscobar** [libigl/libigl#2103] [Q&A]
I've been playing around with MarchingCubes algoritm and it works nice but the input should be a SDF grid.

Since the grid has a size of M·N·L cells and making the size of the cell a half it will take 8 times more time to calculate it, and it becomes really really ineficient for U shapes and others since most of the calculous is just wasted (most of the grids contains nothing and they are not neart to any surface.

Is there any chance to boost the performance by using any kind of octree precomputation? Is there any other way to calculate marching cubes faster?

Thank you!

**Comment by jdumas** [comment]
Look at tutorial 715_MeshImplicitFunction, it uses a sparse voxel grid + dual contouring implementation that doesn't require you to evaluate your SDF on a dense grid.

**Comment by FrankEscobar** [comment]
As far as I know sparse voxel grid is constructed using a function not using a 3d mesh, right?

So it will be invalid for remeshing existing surfaces.

**Comment by FrankEscobar** [comment]
@jdumas it could has sense for binary testing ignoring the neighbours of the grid voxels when the absolute distance is twice the size of the voxel itself? I'm still wondering how to implement that but obviously it sounds that it could be much faster by ignoring the empty spaces, being inside the shape or outside the shape.

Regards.
